src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -197,34 +197,43 @@ public void Start()
                 }
 
                 // STEP 2: Open the socket connection to the remote host
-                var cts = new CancellationTokenSource();
-                var tok = cts.Token;
-                
-                tok.Register(() =>
-                {
-                    _client.Close();
-                });
+                var cts = new CancellationTokenSource(ConnectTimeout);
+                cts.CancelAfter(ConnectTimeout);
 
+                using (cts.Token.Register(() => { _client.Close(); }))
+                {
+                    _client.Dispose();
+                }
                 try
                 {
-                    _client.ConnectAsync(_logic.UrlRequest.Host, _logic.UrlRequest.Port).ContinueWith(t =>
+                    using (cts)
                     {
-                        if (!NetworkTaskSuccessful(t)) {
-                            return;
-                        }
+                        _client.ConnectAsync(_logic.UrlRequest.Host, _logic.UrlRequest.Port).ContinueWith(t =>
+                        {
+                            if (!NetworkTaskSuccessful(t))
+                            {
+                                return;
+                            }
 
-                        _queue.DispatchAsync(StartInternal);
+                            _queue.DispatchAsync(StartInternal);
 
-                        if (tok.IsCancellationRequested)
-                            return;
-                    }, tok);
+                        }, cts.Token);
+                    }
+                }
+                catch (OperationCanceledException)
+                {
+                    // connection timeout
+                    _client.Close();
                 }
-                catch (Exception e) {
+                catch (SocketException)
+                {
+                    _client.Close();
```

Comment:
I'm not sure what the meaning of this part is.  Did you find some information indicating that the client would not be already closed when a `SocketException` happens?  Also `DidClose` is needed to communicate the closure to LiteCore.